lib/std/c/linux.zig

```diff
@@ -82,6 +82,8 @@ pub extern "c" fn sigaltstack(ss: ?*stack_t, old_ss: ?*stack_t) c_int;
 
 pub extern "c" fn memfd_create(name: [*:0]const u8, flags: c_uint) c_int;
 
+pub extern "c" fn ftruncate64(fd: c_int, length: off_t) c_int;
```

Comment:
> I believe the libcs transparently make ftruncate call the correct syscall.

glibc predates the 64-bit era by quite some time, the "bare" syscalls may be routed to the 64-bit variant (see #3153) if some defines are set.
Musl took a different approach and decided to only support the 64-bit versions, but we should use the `64`-prefixed versions whenever possible.

Reply:
In this case we could use `std.c.versionCheck` (which maybe should be renamed, but it will return `true` for musl and check against glibc's version) to determine whether to call ftruncate64 or ftruncate.

However, are you sure that is necessary? In the man pages, I see:

```
       The original Linux truncate() and ftruncate() system calls were not de‐
       signed to handle large file offsets.   Consequently,  Linux  2.4  added
       truncate64()  and  ftruncate64()  system calls that handle large files.
       However, these details can be  ignored  by  applications  using  glibc,
       whose  wrapper  functions  transparently  employ the more recent system
       calls where they are available.
```

Reply:
> In this case we could use std.c.versionCheck (which maybe should be renamed, but it will return true for musl and check against glibc's version) to determine whether to call ftruncate64 or ftruncate.

It's not a matter of glibc version, it's a matter of compile-time flags we cannot (easily) detect. Picking the `64` version on glibc (or on Musl, it doesn't really matter there) is needed to make sure the `off_t` for the libc is wide enough.

Reply:
Oh, I see that musl provides an `ftruncate64` alias. OK, so I think your original code regarding ftruncate vs ftruncate64 is correct (which makes sense since it passed the tests, and we test against musl libc). Sorry for the noise.